### PR TITLE
2504 Patient list loading issues

### LINKED ISSUE #2504: Patient list loading issues
There are a couple of issues related to the patient list:
- [x] Pagination is not working: the full list of patient is always loaded
- [x] The patient document data is not needed (?) and can be removed to reduce the amount of loaded data
- [x] The patient list is loaded twice with slightly different parameters:
image
And with different query params:
image

- [ ] The remote server seems to use quite a lot of cpu when loading the full patient list (sqlite). Profile what is happening and how it can be improved. (This can be moved into another issues)

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #2504 

# 👩🏻‍💻 What does this PR do? 
- Fixes pagination
- Removes `document.data` from Generic Patient Fragment and created a new fragment to use every where other than patient list view that has document data in it.

# 🧪 How has/should this change been tested? 
- [ ] Check if pagination is working correctly
- [ ] Patient list view should load a little faster
- [ ] Check that searches still work (can find one in program enrolment -> mother field)

### CODE REVIEW

File: client/packages/common/src/types/schema.ts [hunk 1]
```diff
@@ -4014,6 +4014,8 @@ export type QueriesStocktakeByNumberArgs = {
 
 
 export type QueriesStocktakeLinesArgs = {
+  filter?: InputMaybe<StocktakeLineFilterInput>;
```
Comment (roxy-dao) [inline client/packages/common/src/types/schema.ts:4017]: Not sure why this was missing when I generated..

**Review by mark-prins — CHANGES_REQUESTED** [review]
This will mean that if you filter the patient list, the csv export will only export the filtered view - which makes more logical sense, but is not like anywhere else on the site!

the main problem though is that it will use the current pagination, so the export will only export the current page of results, but we want it to export all results, hence using the `listAll` api call.

I tried replicating and I only have one patient API query being made - do you have a way to replicate the issue?

**Comment by clemens-msupply** [comment]
@mark-prins can we only do the query when the export is done? I send you a datafile where I see the issue. However, the datafile is quite simple, so I am not sure why you are not seeing it...

I am getting two calls to patients when navigating to the patient list.

**Comment by roxy-dao** [comment]
> This will mean that if you filter the patient list, the csv export will only export the filtered view - which makes more logical sense, but is not like anywhere else on the site!
> 
> the main problem though is that it will use the current pagination, so the export will only export the current page of results, but we want it to export all results, hence using the `listAll` api call.
> 
> I tried replicating and I only have one patient API query being made - do you have a way to replicate the issue?

Ah... true, I forgot to take pagination into account, but when I download the csv it loads all the patients regardless of pagination... the ''bug'' @clemens-msupply was stating in the issue in in all list views with the 'Export to CSV' button...

**Comment by mark-prins** [comment]
only when the button onclick is called - that's why we are using the mutation. can't see any reason in the code why it would be called, nor any reason why it is different for the patients. the same pattern has been used everywhere else.

**Comment by roxy-dao** [comment]
> only when the button onclick is called - that's why we are using the mutation. can't see any reason in the code why it would be called, nor any reason why it is different for the patients. the same pattern has been used everywhere else.

It happens everywhere...

**Comment by clemens-msupply** [comment]
Yes right, listAll is a mutation and I also still see both queries. 

I am suspecting that the two queries happening because the filter/sort parameters are changing during the initial rendering. Not sure how to best avoid it.

The pagination and potentially the removal of the unneeded document data are IMHO the things that we should tackle first. Loading a little bit of data twice might then be acceptable if we can't find an easy solution for the double load.

**Comment by mark-prins** [comment]
The problem is the query params handling - if you refresh the page, you only get one patients query made.
If you click on 'patients' in the nav, you'll see two. The first has no 'sort' option, the second does.

File: client/packages/programs/src/JsonForms/components/Search/SearchWithUserSource.tsx [hunk 2]
```diff
@@ -38,12 +41,16 @@ export const SearchWithUserSource = (
   const t = useTranslation('programs');
 
   const isPatientSelected = !!data?.id;
-
   const searchFilter = !isPatientSelected
     ? createSearchFilter(options?.searchFields, data)
     : undefined;
 
-  const { results, error: queryError } = usePatientSearchQuery(searchFilter);
+  const { results, error: queryError, mutateAsync } = usePatientSearchQuery();
```
Comment (roxy-dao) [inline client/packages/programs/src/JsonForms/components/Search/SearchWithUserSource.tsx:48]: Changed to use patientSearch hook instead... feels more fitting...

**Review by mark-prins — APPROVED** [review]
👍  Works for me!

**Review by clemens-msupply — APPROVED** [review]
Works great thank you!

**Pull request merged**